### ISSUE TITLE
Buggy work with Undercore's debounce

### ISSUE THREAD

**Issue by ogonkov** [NYTimes/backbone.stickit#248]
I've tired to update model value after user stopped his input, but it seems working very strange.

Here is a little demo http://jsbin.com/rulula/4/

When i wrap `onSet` callback into `_.debounce` it's working pretty tricky.

I have add some `setInterval` for demonstration, because when `textarea` lost focus, value is updating as expected. But it seems very strange for me, because it should update after 250 ms of debounce.

What i'm doing:
1. Click on `textarea`;
2. Start some input ('12345' for example);
3. Press 'spacebar';
4. Preview should show '12345';
5. Now select some of your text with keyboard (SHIFT button);
6. Remove selected text.

This action took less than 250ms.

What i expect:
Preview area should show my rest text (ie get it from model).

What happens:
It will keep show previous text. It's looks like something happen with debounced function and it's waiting for one more call. If i blur textarea (click somewhere outside of textarea), it's update model as expected and updates preview.

If i start invoke callback without debounce, everything is OK, so problem somewhere with working stickit & debounce together.

What i'm doing wrong?

**Comment by euge** [comment]
Thanks for all the ideas here, I came up with a slightly differently solution involving using `silent:true`
```javascript
{
  bindings: {
    ":text" : {
      observe: "attrName",
      setOptions: {
        silent: true
      },
      onSet: "_setSomething"
    }
  },

  initialize: function() {
    var model = this.model;
    this._lazyChange = _.debounce(function() {
      model.trigger("change");
    }, 300);
  },

  _setSomething: function(val, options) {
    this._lazyChange();
    return val;
  }
}
```
With this approach, the new attributes are set in realtime, but the change notification happens 300ms when the user input stops.

**Comment by spectras** [comment]
That should work, but be a bit error-prone if you have other devs on the project (or yourself, in a few years). Understanding why update events fire 300ms late if you don't know this trick will let people scratching their head for a while. It may even cause race conditions if something else updates the attribute in between `_setSomething` and the actual event.
Your call. By the way, to mimic all events, you probably want to trigger a `change:attrname` event as well ^^

**Comment by akre54** [comment]
Agreed with @spectras. This is definitely hackier than it should be. If there's something Stickit can do to improve I'm all ears.

(Could we maybe allow you to set `val` in the `config` object so we wouldn't have to care about the return value? or some other way of avoiding the syncronicity of onSet?)

**Comment by spectras** [comment]
I guess you could document `config.set` and make it an official part of the API? It does the job, but last time I looked it was not documented. From [this function](http://nytimes.github.io/backbone.stickit/docs/annotated/#section-32) it looks it is intended for overriding the actual setting of a value into the model.
